shims/spark33/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala

```diff
@@ -99,17 +101,25 @@ class HiveFileFormat(fileSinkConf: FileSinkDesc)
     if ("true" == sparkSession.sparkContext.getLocalProperty("isNativeApplicable")) {
       val nativeFormat = sparkSession.sparkContext.getLocalProperty("nativeFormat")
       val tableOptions = tableDesc.getProperties.asScala.toMap
-      val isParquetFormat = nativeFormat == "parquet"
-      val compressionCodec = if (fileSinkConf.compressed) {
-        // hive related configurations
-        fileSinkConf.compressCodec
-      } else if (isParquetFormat) {
-        val parquetOptions =
-          new ParquetOptions(tableOptions, sparkSession.sessionState.conf)
-        parquetOptions.compressionCodecClassName
-      } else {
-        val orcOptions = new OrcOptions(tableOptions, sparkSession.sessionState.conf)
-        orcOptions.compressionCodec
+      val compressionCodec = nativeFormat match {
+        case "parquet" if fileSinkConf.compressed =>
+          // MapredParquetOutputFormat use the `ParquetOutputFormat.COMPRESSION` as
+          // the compression codec. Currently, we only support the parquet native
+          // writer for Spark3.2/3.3 code path.
```

Comment:
Is the comment right? @JkSelf

Reply:
comment about code path has been removed.